src/types/UiaTextRangeBase.cpp

```diff
@@ -372,22 +620,172 @@ try
+        pRetVal->lVal = base::ClampedNumeric<long>(_pData->GetAttributeColors(attr).second & 0x00ffffff);
+        return _getAttrVerificationFn(attributeId, *pRetVal);
+    }
+    case UIA_FontWeightAttributeId:
+    {
+        // The font weight can be any value from 0 to 900.
+        // The text buffer doesn't store the actual value,
+        // we just store "IsBold" and "IsFaint".
+        // Source: https://docs.microsoft.com/en-us/windows/win32/winauto/uiauto-textattribute-ids
+        pRetVal->vt = VT_I4;
+        if (attr.IsBold())
+        {
+            pRetVal->lVal = FW_BOLD;
+        }
+        else
+        {
+            pRetVal->lVal = FW_NORMAL;
+        }
+        return _getAttrVerificationFn(attributeId, *pRetVal);
+    }
+    case UIA_ForegroundColorAttributeId:
+    {
+        pRetVal->vt = VT_I4;
+        const auto x{ _pData->GetAttributeColors(attr).first };
+        const auto y{ x & 0x00ffffff };
+        pRetVal->lVal = y;
+        //pRetVal->lVal = base::ClampedNumeric<long>(_pData->GetAttributeColors(attr).first & 0x00ffffff);
+        return _getAttrVerificationFn(attributeId, *pRetVal);
+    }
+    case UIA_IsItalicAttributeId:
+    {
+        pRetVal->vt = VT_BOOL;
+        pRetVal->lVal = attr.IsItalic();
+        return _getAttrVerificationFn(attributeId, *pRetVal);
+    }
+    case UIA_StrikethroughStyleAttributeId:
+    {
+        pRetVal->vt = VT_I4;
+        pRetVal->lVal = static_cast<short>(attr.IsCrossedOut() ? TextDecorationLineStyle_Single : TextDecorationLineStyle_None);
+        return _getAttrVerificationFn(attributeId, *pRetVal);
+    }
+    case UIA_UnderlineStyleAttributeId:
+    {
+        pRetVal->vt = VT_I4;
+        if (attr.IsDoublyUnderlined())
+        {
+            pRetVal->lVal = static_cast<short>(TextDecorationLineStyle_Double);
+        }
+        else if (attr.IsUnderlined())
+        {
+            pRetVal->lVal = static_cast<short>(TextDecorationLineStyle_Single);
+        }
+        else
+        {
+            pRetVal->lVal = static_cast<short>(TextDecorationLineStyle_None);
+        }
+        return _getAttrVerificationFn(attributeId, *pRetVal);
+    }
+    default:
+        // This attribute is not supported.
+        pRetVal->vt = VT_UNKNOWN;
+        UiaGetReservedNotSupportedValue(&pRetVal->punkVal);
+        return nullptr;
+    }
+}
+
+IFACEMETHODIMP UiaTextRangeBase::GetAttributeValue(_In_ TEXTATTRIBUTEID attributeId,
                                                    _Out_ VARIANT* pRetVal) noexcept
 {
     RETURN_HR_IF(E_INVALIDARG, pRetVal == nullptr);
+    VariantInit(pRetVal);
 
-    if (textAttributeId == UIA_IsReadOnlyAttributeId)
+    // AttributeIDs that require special handling
+    switch (attributeId)
+    {
+    case UIA_FontNameAttributeId:
+    {
+        pRetVal->vt = VT_BSTR;
+        pRetVal->bstrVal = SysAllocString(_pData->GetFontInfo().GetFaceName().data());
+        UiaTracing::TextRange::GetAttributeValue(*this, attributeId, *pRetVal);

```

Comment:
This has no chance of throwing, right? I want to make sure we don't leak out the `BSTR` if this line can fail and be caught by someone.

Reply:
Correct. It's `noexcept`